server.go

```diff
@@ -66,42 +64,77 @@ type httpHeaderOpts struct {
 	groupsHeader string
 }
 
+type userHeaderFn func(user *User) string
+
+type userHeaderHelper struct {
```

Comment:
open to ideas for a better name here than "Helper"

Reply:
I ran through some alternatives in my mind (Util, Middleware, Handler) and all of them fit even less. Helper is probably fine for now.

Reply:
SGTM, KISS :)